homeassistant/components/glances/sensor.py

```diff
@@ -41,7 +41,7 @@
     'process_thread': ['Thread', 'Count', 'mdi:memory'],
     'process_sleeping': ['Sleeping', 'Count', 'mdi:memory'],
     'cpu_use_percent': ['CPU used', '%', 'mdi:memory'],
-    'cpu_temp': ['CPU Temp', TEMP_CELSIUS, 'mdi:thermometer'],
+    'cpu_temp': ['CPU Temp', '', 'mdi:thermometer'],
```

Comment:
This makes no sense. You are removing a unit of measurement?

Reply:
This is not predictable for users, we should not merge this.

Reply:
Removing the default unit of measurement, as we can't determine the correct one until we get a response from the API. Setting this to either `TEMP_CELSIUS` or `TEMP_FAHRENHEIT` by default doesn't make sense because we don't know the correct unit until the actual API call down below

Reply:
Ah, I get it now. 

Then why would we not always use `sensor['unit']` for everything?

Reply:
Out of the things we currently monitor in Glances, `sensor['unit']` is only present for `cpu_temp`

Reply:
Is `sensor['unit']` going to be equal to our constants `TEMP_CELSIUS` and `TEMP_FAHRENHEIT` or do we need to convert it ?

Reply:
Unfortunately no, `sensor['unit']` will be either `C` or `F`. I can prepend it with a `°` to keep the end result the same:
`unit_of_measurement = '°' + str(sensor['unit'])`
`°C or °F`

Reply:
I would prefer mapping it: `unit_of_measurement = {'C': CONST_CELSIUS, 'F': CONST_FAHRENHEIT}.get(sensor['unit'])

Reply:
https://github.com/home-assistant/home-assistant/pull/6433#discussion_r106589509

The above comment gives me the impression that returning `°C` or `°F` as `unit_of_measurement` will cause HomeAssistant to auto convert the temperature to the user's `preferred_units`? 

So if:
`preferred_units = imperial` and we return `unit_of_measurement` as `CONST_CELSIUS`, HASS will convert it to the equivalent degrees in Fahrenheit, correct? Is there a way to prevent this?

Reply:
That is correct and there is not currently a way to prevent this. If we want to change this, it should be changed at a Home Assistant architecture level. We should not add workarounds to integrations.